### PR TITLE
Price Estimation Fee application - hook logic and proxy

### PR DESCRIPTION
**Pull request by W3stside**
Part of #32 

Replaces #33 

- [x] Initial logic that overrides useDerivedSwapInfo and calculates exactIn/Out values w/tip injected (this PR)
- [x] Mock async `getFee` API endpoint to await on and get fake fee value #35 
- [ ] Implement 'real' `getFee` endpoint can be done soon

Logic:
Proxy our trade object and apply fee on amounts after calculation. Doesn't recalcualte anything with fee values as before, just overrides the `input/outputAmount` with our fee wrapped method.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr48--gpv2swap.review.gnosisdev.com)**: Testing web app

File: src/custom/state/swap/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+function parseTokenAmountURLParameter(urlParam: any): string {
+  return typeof urlParam === 'string' && !isNaN(parseFloat(urlParam)) ? urlParam : ''
+}
+
+function parseIndependentFieldURLParameter(urlParam: any): Field {
+  return typeof urlParam === 'string' && urlParam.toLowerCase() === 'output' ? Field.OUTPUT : Field.INPUT
+}
+
+const ENS_NAME_REGEX = /^[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&/=]*)?$/
+const ADDRESS_REGEX = /^0x[a-fA-F0-9]{40}$/
```
Comment (Velenir) [inline src/custom/state/swap/hooks.ts:275]: Faaancy, I'm gonna trust you on that ENS regex

you can drop `a-zA-Z` duplication if you use `/.../i` ignore case flag

File: src/custom/state/swap/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+function validatedRecipient(recipient: any): string | null {
```
Comment (Velenir) [inline src/custom/state/swap/hooks.ts:276]: can recipient be typed, or too troublesome?

**Review by Velenir — APPROVED** [review]
I see you reached for the Big Guns (Proxy)

File: src/custom/state/swap/hooks.ts (hunk 1 above)
Comment (W3stside) [reply src/custom/state/swap/hooks.ts:275]: lol i didnt write any of that :sweat_smile:

**Comment by W3stside** [comment]
i need to remove some code that's duped from this file, forgot the bottom exports woops

**Review by anxolin — COMMENTED** [review]
I believe this PR is pending the review we discussed in our last sync.
Any progress on that? I've just made a similar comment in https://github.com/gnosis/gp-swap-ui/issues/15

**Comment by W3stside** [comment]
closing, superseded by #103

**Pull request closed without merging**